mpop_kiosk/main.qml

```diff
@@ -139,6 +152,17 @@ ApplicationWindow {
         onLastRfidReadChanged: {
             lastRfidRead = rfidReader.lastRfidRead;
             console.log("(QML) Last RFID read: " + lastRfidRead);
+
+            // Case: 2 :: free the latest current Tag when new Tag scan and Kiosk is in Exit mode.
+
+             if (kioskConfig.kiosk_mode == window.const_KIOSK_MODE_EXIT) {
+                userProfile.freeTag(function(err,result){
+                    if (err) {
+                        console.log("Error calling  freeTag: " + err.message);
+                    }
+                })
+               }
+
```

Comment:
At this point, the value of lastRfidRead is the new tag. We should move this up to line 153, then call it with the previous tag, not the new one. 

```
var oldTag = window.lastRfidRead;
websocket.callRemoteMethod("freeTag",[oldTag], ...
```

Reply:
ok.. let me update